api/jobs.py

```diff
@@ -172,7 +172,7 @@ def retry_job(db, j, force=False):
         log.info('permanently failed job %s (after %d attempts)' % (j['_id'], j['attempt']))
 
 
-def generate_formula(algorithm_id, i):
+def generate_formula(algorithm_id, i, job_id=None):
```

Comment:
I thought about that, but I wasn't sure I wanted to make an existing job record and id a prerequisite for generating this kind of static content.

The origin stuff will pick up on the engine as a fallback origin if no job parameter is present, so I'd like to keep this optional unless someone has a strong opinion on it.